Go/Go.sublime-syntax

```diff
@@ -873,67 +944,89 @@ contexts:
 
         - include: pop-before-nonblank
 
+  pop-func-method-receiver:
+    - match: \(
+      scope: punctuation.section.parens.begin.go
+      set: pop-func-method-receiver-inner
+    - include: pop-before-nonblank
+
+  pop-func-method-receiver-inner:
+    - include: pop-on-paren-end
+    - include: match-type-operators
+    - include: match-namespace
+
+    # The lookahead is too special-case. TODO improve.
+    - match: '{{ident_blank}}(?={{noise}}(?:\*|{{ident}}))'
+      scope: variable.language.blank.go
+    - match: '{{ident}}(?={{noise}}(?:\*|{{ident}}))'
+      scope: variable.parameter.go
+
+    - match: '{{ident}}'
+      scope: storage.type.go
+      push: pop-type-parameter-list-simple
+
+    - include: match-any
+
   pop-func-signature:
-    - include: match-comments
-    - match: \b_\b
+    - match: (?=\S)
+      set: [pop-func-parameter-and-return-lists, pop-type-parameter-list, pop-func-name]
+
+  pop-func-name:
+    - match: '{{ident_blank}}'
       scope: variable.language.blank.go
-      set: pop-func-parameter-and-return-lists
+      pop: true
     - match: '{{ident}}'
       scope: entity.name.function.go
-      set: pop-func-parameter-and-return-lists
+      pop: true
     - include: pop-before-nonblank
 
+  pop-func-parameter-and-return-lists:
+    - match: (?=\()
+      set: [pop-func-return-signature, pop-func-parameter-list]
+    - include: pop-before-nonblank
+
+  pop-func-return-signature:
+    - include: pop-before-terminator
+    - match: (?=\()
+      set: pop-func-parameter-list
+    - match: (?=\S)
+      set: pop-type
```

Comment:
The way `pop-type` is implemented the following would work, too, and avoid one extra context switch.

```suggestion
    - include: pop-type
```